packages/shared/src/hooks/useAccountEmailFlow.ts

```diff
@@ -34,35 +47,58 @@ type EmailFlow = AuthFlow.Recovery | AuthFlow.Verification;
 function useAccountEmailFlow({
   flow,
   flowId,
-  isQueryEnabled,
   onSuccess,
   onVerifyCodeSuccess,
   onError,
   onTimerFinished,
 }: UseAccountEmailProps): UseAccountEmail {
-  const [resentCount, setResentCount] = useState(0);
+  const [activeFlow, setActiveFlow] = useState(flowId);
+  const [autoResend, setAutoResend] = useState(false);
   const { timer, setTimer, runTimer } = useTimer(onTimerFinished, 0);
 
+  const existingEnabled = flow === AuthFlow.Verification && !flowId;
+
+  const { data: existingFlow, isLoading: existingFlowLoading } = useQuery(
+    ['existing_flow'],
+    getVerificationSession,
+    {
+      ...disabledRefetch,
+      enabled: existingEnabled,
+      retry: false,
+    },
+  );
+
+  if (existingFlow?.result?.flow_id && !activeFlow) {
+    setActiveFlow(existingFlow.result.flow_id);
+  }
+
+  const enabled =
+    flow === AuthFlow.Recovery
+      ? true
+      : !existingFlowLoading && !activeFlow && !autoResend;
+
   const { data: emailFlow } = useQuery(
-    [{ type: flow, sentCount: resentCount }],
+    [{ type: flow }],
     ({ queryKey: [{ type }] }) =>
-      flowId
-        ? getKratosFlow(AuthFlow.Recovery, flowId)
-        : initializeKratosFlow(type),
+      flowId ? getKratosFlow(flow, flowId) : initializeKratosFlow(type),
     {
-      ...(flowId
-        ? { ...disabledRefetch, retry: false }
-        : { enabled: isQueryEnabled }),
+      ...disabledRefetch,
+      enabled,
     },
   );
 
+  if (emailFlow?.id && !activeFlow) {
+    setActiveFlow(emailFlow.id);
+    setAutoResend(true);
+  }
```

Comment:
This also needs to go to `useEffect` or potentially you can just memoize and avoid extra state since it just depends on emailFlow value.

Reply:
Good point, I kind of assumed the values are available on the re-render which seems to be enough.
What case would they not be up to date then? (where you need the useEffect) (kind of feels like that would just trigger yet another re-render?)

Reply:
Setting state during render will do another render which can then do another and another so you can end up in infinite loop really fast. This is why it is not recommended. Usually it can be replaced with:
- `useEffect` + `useState`
- `useMemo` - just compute new values from current state, which I think you can do in this case.

Reply:
Alright let me memo it for now :) 
I don't like to many useEffects lol